nixops/ssh_util.py

```diff
@@ -299,5 +299,28 @@ def run_command(self, command, flags=[], timeout=None, logged=True,
             else:
                 return res
 
+    def try_ssh(self, user=None, timeout=-1):
+        try:
+            self.run_command('true', timeout=1, user=user, flags=['-q'])
+            return True
+        except nixops.ssh_util.SSHConnectionFailed:
+            return False
+
+
+    def wait_for_ssh(self, user=None, attempts=-1, timeout=5, callback=None):
+        """Wait until the remote's SSH is up."""
+        n = 0
+        while True:
+            try:
+                self.run_command('true', timeout=5, user=user, flags=['-q'])
+                return True
+            except nixops.ssh_util.SSHConnectionFailed:
```

Comment:
Why not re-use `try_ssh` here instead of doing roughly the same thing here again?